let res,GeneralError;_2a6‍.x([["handleErrors",()=>handleErrors]]);_2a6‍.w("express/lib/response",[["default",["res"],function(v){res=v}]]);_2a6‍.w("../utils/errors",[["GeneralError",["GeneralError"],function(v){GeneralError=v}]]);


       const handleErrors = async (err, req, res, next) => {
    if (err instanceof GeneralError) {
        const code = err.getCode();
        return res.status(code).json({
            name: err.name,
            message: err.message
        });
    }
    return res.status(500).json({
        name: "Internal Server Error",
        message: err.message
    })
}